feat: landing page

ashleysyg opened the pull request:
This is a test for anima code generation for an entire figma frame. The figma frame I'm testing is the landing page. I will compare the code quality, fidelity to the design, and how efficient it is to tweak the code to completion vs. in the feat-carousel page where I tested individual component code generation one at a time. If this works better, I'll create separate PRs for the different sections of the landing page for easier review.

Commit:
chore: ignore .anima folder because its large and unnecessary to version control

Commit:
added landing page components to home

Commit:
added base components for landing page

Commit:
added ui components and installed libraries for thme

Commit:
chore: reorganize header components

Commit:
testing components from anima in landing page

Commit:
fix: position of events

Commit:
fix: imports for logo name changes and added baseURL to tsconfig to fix imports

Commit:
fix: resolves client vs server conflicts between

Commit:
fix: update tailwind config with new ccv color style guide, refactored and fixed styling for status banner, made good status primary ccv green

Commit:
style: brown banner style fixes, logo size, alignment and spacing, work with us button

Commit:
fix: make most style fixes to footer

Commit:
feat: source sans pro is default font

Commit:
feat: background gradient for footer

Commit:
feat: improve styling of footer, set full width for the landing page and all components

Commit:
chore: remove relative imports

Commit:
chore: button state colors - still not right

Commit:
chore: move project content inside of container flanked by left and right arrows

Commit:
fix: broken still. hero title animation introduces undefined

Commit:
wip: merge 2 navbars

Commit:
feat: adds impact stats

ashleysyg commented:
To Do:

- [ ] chore: update tailwind config with styling from figma that have not yet been added (colors, margins, padding, mobile vs desktop flex layout, etc.)
- [ ] feat: resizing components with window size
- [ ] chore: page margins
- [ ] fix: hero title animation 
- [ ] feat: events section updates (title, button, etc.)
- [ ] rename components to match figma names

eldu commented on components/AnimatedTitle.tsx:
TODO: fix in another PR

eldu commented on components/Hero.tsx:
```diff
@@ -0,0 +1,40 @@
+
+export const Hero: React.FC<HeroProps> = ({ onViewEventsClick }) => {
+  return (
+    <section className="w-full h-[810px] bg-[url(https://c.animaapp.com/VOhWj8ET/img/adobestock-443655134-1.png)] bg-cover bg-center relative">
```
This needs to be our image. TODO: go through and change the anima hosted images to our own images

eldu commented on components/Hero.tsx:
```diff
@@ -0,0 +1,40 @@
+  onViewEventsClick: () => void;
+}
+
+export const Hero: React.FC<HeroProps> = ({ onViewEventsClick }) => {
```
Think about ways to generalize the hero component to use for our other heros

eldu commented on components/HeroCard.tsx:
```diff
@@ -0,0 +1,52 @@
+import React from "react";
+import { Card, CardContent } from "@/components/ui/card";
+
+const heroCards = [
```
to generalize the component, it would be nice if it took this in as a prop

eldu replied:
opens up conversation about structuring data for the website

eldu commented on components/ImpactBanner.tsx:
```diff
@@ -0,0 +1,42 @@
+import React from "react";
+import { Card, CardContent } from "@/components/ui/card-impact";
+
+export const ImpactBanner = (): JSX.Element => {
```
maybe we rename this on the figma&here to be more general like KPI Banner? But let this live for now

eldu commented on components/ImpactBanner.tsx:
```diff
@@ -0,0 +1,42 @@
+
+export const ImpactBanner = (): JSX.Element => {
+  // Data for impact metrics
+  const impactMetrics = [
```
make as a prop

eldu commented on components/ui/card-impact.tsx:
I think this can be simplified a LOT

Commit:
create folders for global images and logos

gtdang committed:
fix: vertically align cards

gtdang committed:
fix: center the component

gtdang committed:
fix: changed fixed width to hug the cards

Commit:
public/static/images not yet working

Commit:
remove ; for consistency

gtdang committed:
Merge pull request #45 from brown-ccv/feat-herocard-formatting

feat: herocard formatting

Commit:
fix: hero image

Commit:
chore: replace all animaapp screenshots with PNGs downloaded from figma

ashleysyg committed:
Merge branch 'feat-landing' into chore-images

ashleysyg committed:
Merge pull request #47 from brown-ccv/chore-images

Chore images

Commit:
chore: change component names to match figma names

Commit:
chore: remove duplicate component and impact banner styling

Commit:
chore: add center or left alignment prop to section headers

Commit:
refactor: create small profile card for groups/depts/labs

Commit:
chore: add vertical padding between hero and impact banner

Commit:
refactor: opportunities to use general card component instead of LinkCard

Commit:
chore: remove outlien adn shadow around carousel images

Commit:
chore: remove unused components

Commit:
chore: adjust spacing

Commit:
fix: use magic ui for animated hero title

Commit:
chore: update classnames using sources sans/serif fonts

Commit:
chore: remove custom colors containing "default" - this is not the correct naming schema

Commit:
chore: remove custom "default" from text class

ashleysyg committed:
Merge pull request #51 from brown-ccv/fix-animated-title

fix: use magic ui for animated hero title

ashleysyg committed:
Merge branch 'feat-landing' into refactor-fonts

ashleysyg committed:
change organization for PPHC

Co-authored-by: Ellen Duong <shi.ellenduong@gmail.com>

Commit:
chore: fix ref to events, not featured carousel (still needs to be refactored in diff PR)

ashleysyg committed:
Merge pull request #49 from brown-ccv/refactor-cards

refactor cards

Commit:
fix merge conflicts

Commit:
Merge branch 'refactor-fonts' of https://github.com/brown-ccv/ccv-website into refactor-fonts

ashleysyg committed:
Merge pull request #53 from brown-ccv/refactor-fonts

Refactor fonts

gtdang committed:
refactor: auto-reformat

gtdang committed:
fix: add use client directive at the top of navbar and featured carousel files

gtdang committed:
feat: added utility function to scroll to target by document id

gtdang committed:
feat!: changed hero to to take children prop

gtdang committed:
feat: added a HeroHome component for the homepage hero

This was separated because passing in scrolling handlers must be client-side and the page.tsx is server-side.

gtdang committed:
refactor: updated page to remove ClientLayout and import individual components

Improves the clarity of page elements.

gtdang committed:
refactor: deleted ClientLayout file

gtdang committed:
feat: added image prop to Hero

gtdang committed:
refactor: added image prop to Hero call

gtdang committed:
refactor: update layout

refactor: update layout

gtdang committed:
feat: changed footer to ChatCCV footer

feat: changed footer to ChatCCV footer

gtdang committed:
build: added react-icons package

gtdang committed:
refactor: combined divs

gtdang committed:
refactor: make hero height smaller

gtdang committed:
Merge pull request #58 from brown-ccv/refactor-hero

refactor: hero

gtdang committed:
refactor: added separator boarder, resized, and regrouped components

gtdang committed:
refactor: remove old events section header

gtdang committed:
refactor: changed colors and rounding of events toggle button

gtdang committed:
refactor: reordered toggle button selections

gtdang committed:
refactor: added margin and color to events div

gtdang committed:
refactor: adjust padding of events section

gtdang committed:
refactor: add spacing between toggle options

gtdang committed:
feat: added heading Card for event section

gtdang committed:
refactor: changed padding

gtdang committed:
Merge pull request #59 from brown-ccv/33-feat-footer

feat footer

gtdang committed:
refactor: Update padding

Co-Authored-By: Ellen Duong <shi.ellenduong@gmail.com>

Commit:
chore: remove tailwind config genereated by anima

Commit:
chore: add colors from figma style guide 03.

Commit:
chore: clean up color overrides

Commit:
fix: page margins and color refs

gtdang committed:
refactor: changed font

Commit:
refactor: button uses more globally aplicable variant-styles

Commit:
wip: refactoring badge and featured carousel to use new global styling setup

gtdang committed:
refactor: add className prop to CCVBars component and default bottom margin

gtdang committed:
refactor: adjusted margins where CCVBars were used

gtdang committed:
refactor: replace BrownLogo component with color OIT png logo in BrownBanner component

gtdang committed:
Merge pull request #61 from brown-ccv/refactor-header

refactor: header banner

Commit:
wip: defined button, badge, card variants in a single ui file not in individual components

Commit:
feat: util function to cycle through 10 diff colors for badges (tags)

Commit:
refactor: herocard to use newly defined card variant

Commit:
chore: move getcolorfortag into utils

Commit:
chore: replace primary/secondary with keppel/sunglow

Commit:
fix: sunglow-blue-# to blue-#

Commit:
fix: figma guide colors updated and sane names

Commit:
chore: rename buttons and add black styles (wip)

gtdang committed:
refactor: font change

gtdang committed:
refactor: adjust layout and styling of EventSection component on small screens

gtdang committed:
refactor: update background color of weekly calendar view

gtdang committed:
Merge branch 'feat-landing' into 48-feat-events-section

gtdang committed:
refactor: update import paths in EventSection component

gtdang committed:
Merge pull request #62 from brown-ccv/48-feat-events-section

48 feat events section

Commit:
chore: button references

Commit:
chore: impact banner flex behavior for small screens

Commit:
merge fix

Commit:
refactor: ccv bar margins

gtdang committed:
refactor: move navbar out of HeroHome

gtdang committed:
refactor: adjust Hero component height and HeroHome layout for improved responsiveness

gtdang committed:
refactor: updated footer background and text colors

gtdang committed:
refactor: define RouteItem and NavigationProps interfaces for improved typescript

ashleysyg committed:
clean: unused and outdated ui components

ashleysyg committed:
Remove SectionHeader in merge resolution

ashleysyg committed:
fix: missing hash from custom navbar color

ashleysyg committed:
feat: add and fix all button variants

gtdang committed:
refactor: replace SVG icons with react-icons

gtdang committed:
chore: deleted old Navbar component

gtdang committed:
refactor: renamed NavbarAnima as Navbar

gtdang committed:
Merge branch '37-global-styling' into footer-background-color

ashleysyg committed:
feat: update buttons that need to override default variants

gtdang committed:
Merge branch '37-global-styling' into refactor-nav

ashleysyg committed:
chore: change primary and secondary button shades

ashleysyg committed:
Merge pull request #63 from brown-ccv/37-global-styling

global styling

ashleysyg committed:
Merge pull request #72 from brown-ccv/footer-background-color

refactor: updated footer background and text colors

ashleysyg committed:
Merge branch '37-global-styling' into refactor-nav

ashleysyg committed:
wip: button default text size and adjustments for diff text lengths

gtdang committed:
Merge pull request #74 from brown-ccv/footer-background-color

Footer background color

gtdang committed:
Merge pull request #73 from brown-ccv/refactor-nav

Refactor nav

ashleysyg committed:
tofix: search button

ashleysyg committed:
Merge branch 'feat-landing' into fix-colors

ashleysyg committed:
fix: search icon button

ashleysyg committed:
fix search button expansion in another PR

ashleysyg committed:
fix: buttons are auto width with padding on either side of texxt

ashleysyg committed:
chore: clean up color overrides in buttons

ashleysyg committed:
feat: events window resizing behavior fixed so cards change size, the padding between cards stay consistent, and teh flex behavior works as intended

ashleysyg committed:
chore: remove comments

ashleysyg committed:
chore: add bottom margin so there is spacing between card and impact banner

ashleysyg committed:
feat: hero height slightly taller

ashleysyg committed:
Merge pull request #76 from brown-ccv/fix-colors

fix: button default text size and adjustments for diff text lengths

ashleysyg committed:
Merge branch 'feat-landing' into fix-hero-layout

ashleysyg committed:
feat: add center for computation and visualization to hero

ashleysyg committed:
feat: remove horizontal compression from ccv text

ashleysyg committed:
chore: remove top margin from brown banner

ashleysyg committed:
chore: update size of work with us button and remove view events button size override

ashleysyg committed:
Merge pull request #77 from brown-ccv/fix-hero-layout

Fix hero layout

ashleysyg committed:
fix: stats stack into one column at wider window size

ashleysyg committed:
chore: remove comments

ashleysyg committed:
fix: viewport resizing issues

ashleysyg committed:
Merge branch 'feat-landing' into feat-brown-header

ashleysyg committed:
fix: added hero height minimum for viewport resizing so card does not overlap with the hero content

ashleysyg committed:
fix: remove places where components widths are not responsive and could possibly be causing left and right whitespace at edges of viewport

ashleysyg committed:
chore: replace any tailwind classes that might cause left and right page margins

ashleysyg committed:
fix: apply zoom to outermost div in body. adds custom css class for responsive zoom

gtdang committed:
fix: correct hero image URL format

gtdang committed:
fix: update hero section text

gtdang committed:
fix: wrap Hero component in a div for base background color if image fails to load

gtdang committed:
Merge pull request #78 from brown-ccv/feat-brown-header

feat: add center for computation and visualization to hero

ashleysyg committed:
fix merge conflict keeping responsive px

ashleysyg committed:
chore: standardize component x padding

ashleysyg committed:
fix: hero text horizontal spillover

ashleysyg committed:
fix: hero text spillover, button width and flex behavior, consistent top margins for buttons, apply zoom across all screen sizes

ashleysyg committed:
refactor: adds services landing page and moves Header, Navbar, and Hero into layout

ashleysyg committed:
merge fix: (wip) carousel wrapping with zoom needs work

ashleysyg committed:
merge fix: clean out merge comments from components

ashleysyg committed:
refactor: replace css grid with flex, fix mobile view, put chevron buttons on either side of the pagination dots

ashleysyg committed:
chore: increase padding between chevron buttons and pagination dots

ashleysyg committed:
Merge branch 'feat-landing' into refactor-hero

ashleysyg committed:
fix: doulbe hero

ashleysyg committed:
feat: change hero animation to make it faster

ashleysyg committed:
refactor: remove use client from app/page and app/layout, default for every route is no status banner except for landing conditional on path, fix: getopenissues to lib folder so it
is not bundled as client when it needs to be server

ashleysyg committed:
chore: clean up unused components and header structure

ashleysyg committed:
feat: change hero animation to mamke it faster

ashleysyg committed:
refactor: move scroll to new button component and make page server side

ashleysyg committed:
fix: mobile view; feat: hero bg image, hero button styles and formatting

gtdang committed:
refactor: event card blue text title

gtdang committed:
fix: weekly view calendar colors

ashleysyg committed:
refactor: move brown banner, navbar and footer into layout.tsx so it is not hidden in LayoutWithStatusBanner

gtdang committed:
fix: typescript implementations

gtdang committed:
refactor: calendar heading font

gtdang committed:
refactor: calendar weekly fonts

gtdang committed:
fix: use EventCard component to for all views

gtdang committed:
refactor: improve CalendarEvent component styling for better readability

gtdang committed:
refactor: adjust CalendarHeading component styles for consistency

gtdang committed:
refactor: update calendar view and tooltip styles for improved readability and consistency

gtdang committed:
fix: set body zoom level to 1

gtdang committed:
refactor: limit to card view for small screens

gtdang committed:
refactor: moved card events to component UpcomingEvents to reduce redundancy

ashleysyg committed:
fix: status banner is green when all operational, remove redundant children

gtdang committed:
refactor: encapsulated ToggleButton component to improve readability

gtdang committed:
fix: add top margin to EventCard

gtdang committed:
refactor: simplify calendar event color assignment

gtdang committed:
Merge pull request #83 from brown-ccv/refactor-hero

Refactor hero

gtdang committed:
refactor: update DataProps interface to include id, date, and description_long fields

gtdang committed:
refactor: update color classes for better consistency in Calendar components

gtdang committed:
refactor: remove redundant class name from TooltipContent

gtdang committed:
Merge pull request #85 from brown-ccv/fix-event-section-formatting

Fix event section formatting

gtdang reviewed (approved):
Approved!
